mernarezk/assignment3/WordSearchTest.java

```diff
@@ -0,0 +1,36 @@
+package Assignment3;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+
+class WordSearchTest {
+	
+	@Test

```

Comment:
How do you feel about adding some more tests? Like using a grid that you design, with words of your choosing and checking for different edge cases like empty grid, empty dictionary, reusing letters, words on a diagonal, words that could be created (but shouldn't) if skipping cells was allowed etc.
You could also unit test some of your helper methods, or the Dictionary class methods.

Reply:
Very nice! Just for the future I'd say perhaps it would be nice to divide such a method into a few smaller tests, but all looks good! :)